reinforcement_learning/rlclientlib/logger/async_batcher.h

```diff
@@ -63,8 +65,16 @@ namespace reinforcement_learning {
 
   template<typename TEvent>
   int async_batcher<TEvent>::append(TEvent&& evt, api_status* status) {
+    if (_queue.size() > _queue_max_size) {
+      if (_queue_dropping_disable) { // block if queue is full
+        std::unique_lock<std::mutex> lk(_m);
+        _cv.wait(lk, [this] { return _queue.size() < _queue_max_size; });
+      }
+      else { // drop events if queue is full
+        _queue.prune(_pass_prob);
+      }
+    }
     _queue.push(std::move(evt));
```

Comment:
Doesn't this change introduce the side effect of the appended event can never be the one that is pruned?

Reply:
Yes. It looks to me more like a choice than a statistical issue.